### PR TITLE
added external cellpose tutorial notebook

### PR DESCRIPTION
**Pull request by SabrinaRichter**

### CODE REVIEW

File: docs/source/external_tutorials/tutorial_cellpose_segmentation.ipynb [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+   "metadata": {},
+   "source": [
+    "# Nuclei segmentation using cellpose"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This tutorial shows how we can easily use an external segmentation packages in the *Squidpy* framework."
```
Comment (hspitzer) [inline docs/source/external_tutorials/tutorial_cellpose_segmentation.ipynb:14]:
```suggestion
    "In this tutorial we show how we can use the anatomical segmentation algorithm `cellpose` in `squidpy.im.segment` for nuclei segmentation."
```

File: docs/source/external_tutorials/tutorial_cellpose_segmentation.ipynb [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+    "import squidpy as sq\n",
+    "import numpy as np\n",
+    "import matplotlib.pyplot as plt"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Prepare cellpose functionality for *Squidpy*"
```
Comment (hspitzer) [inline docs/source/external_tutorials/tutorial_cellpose_segmentation.ipynb:32]:
```suggestion
    "## Prepare custom segmentation function using cellpose"
```

File: docs/source/external_tutorials/tutorial_cellpose_segmentation.ipynb [hunk 3]
```diff
@@ -0,0 +1,400 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To use the cellpose package, we need to install it with `pip install cellpose`.\n",
```
Comment (hspitzer) [inline docs/source/external_tutorials/tutorial_cellpose_segmentation.ipynb:39]: I would move this up to the general explanation (after line 14). Can you also add one sentence explaining what cellpose is? E.g. "cellpose (link-to-code) is a novel anatomical segmentation algorithm. To use it in this example, we need to install it ..."
For referencing the paper, please use `data-cite`, as done in the tangram tutorial: https://squidpy.readthedocs.io/en/latest/external_tutorials/tutorial_tangram.html 
`**Tangram** (<cite data-cite="tangram">Biancalani et al. (2020)</cite>, [code](https://github.com/broadinstitute/Tangram))`. The reference goes in `docs/source/references.bib` (@michalk8 please correct me if im wrong).

File: docs/source/external_tutorials/tutorial_cellpose_segmentation.ipynb [hunk 4]
```diff
@@ -0,0 +1,400 @@
...
+    "number of pixels per mask `min_size`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Wrap the cellpose model initialization and evaluation into one method that we can then pass on to the segmentation\n",
```
Comment (hspitzer) [inline docs/source/external_tutorials/tutorial_cellpose_segmentation.ipynb:73]: I'd suggest to put these comments (and others like them below) in markdown cells. This renders much prettier I think.

File: docs/source/external_tutorials/tutorial_cellpose_segmentation.ipynb [hunk 5]
```diff
@@ -0,0 +1,400 @@
...
+   ],
+   "source": [
+    "# Load the image and visualize its channels. We will use the first channel for segmentation.\n",
+    "\n",
+    "img = sq.datasets.visium_fluo_image_crop()\n",
+    "crop = img.crop_corner(1000, 1000, size=1000)\n",
+    "\n",
+    "fig, axes = plt.subplots(1, 3, figsize=(10, 20))\n",
+    "for i, ax in enumerate(axes):\n",
+    "    crop.show(\"image\", channel=i, ax=ax)"
```
Comment (hspitzer) [inline docs/source/external_tutorials/tutorial_cellpose_segmentation.ipynb:120]: `img.show` should now have a `channelwise` argument that you can use for this plot. No need to define a separate figure and axes, just `crop.show('image', channelwise=True)`. Its possible that this is only on the `dev` branch for now, but we will merge all of that very soon.

**Review by hspitzer — COMMENTED** [review]
Great, thanks a lot for adding an H&E example! This looks quite good as well I believe! 

Can you also do the following (copying parts of Michals comments on another PR here):
1. add the new tutorial to `docs/source/tutorial.rst`
2. run `tox -e check-docs` (there are some typos)
3. run `tox -e shallow-docs` to see if it renders as expected
6. add a thumbnail image? e.g. for tangram, the NB cell with the image has additional metadata
```
   "metadata": {
    "nbsphinx-thumbnail": {
     "output-index": 1,
     "tooltip": "Integrate the Image Container with Tangram for cell-type deconvolution."
    }
   },
```

**Comment by SabrinaRichter** [comment]
looks good now I think

**Comment by michalk8** [comment]
> The channelwise image plotting is not working, yet. Did not include it, yet.

Could you please open an issue in the main repo? I've also added `segmentation_layer: str, segmentation_alpha: float` to `ImageContainer.show` (see [here](https://github.com/theislab/squidpy/pull/324#issuecomment-838803876)), which plots the segmentation over the original image, but I've seen your plotting the segmentation (also next) to it, as you do in the tutorial - any feedback/wishlist is welcome.

**Comment by giovp** [comment]
thanks @SabrinaRichter !
- Can you also add a `cellpose_env.yml` file with your conda env?
- squidpy needs to be installed with pip so either re-create env or manually change the yml file with the latest squidpy (no pin version), so no worry if version don't match here (it will eventually)
- just have minimal reproducible environment
- you can add the env file in the main folder, we can move it around later
- check out some notes that possibly apply here, froma recent similar notebook: https://github.com/theislab/squidpy_notebooks/pull/63#pullrequestreview-661239846
thanks a lot again 🙏

**Comment by giovp** [comment]
I've also added some comments in the notebook app review. Don't know why they don't appear here

**Comment by SabrinaRichter** [comment]
I think should be fine now.

**Comment by giovp** [comment]
looks really nice! thank you for adressing comments. Couple of remaining points:
- the spellcheck is failing, see errors (and suggestions) here:
image

can you Capitalize Visium, Cellpose etc. and if any of these words is not present in `docs/spelling_wordlist.txt` add it there?
- can you add additional metadata to one (any) of the cells with an image for the image thumbnail? you can check how it's done in Tangram by opening the tangram tutorial in editor node and ctrl+F `nbsphinx-thumbnail`. For tangram for instance, it looks like this:
```
"metadata": {
    "nbsphinx-thumbnail": {
     "output-index": 1,
     "tooltip": "Integrate the Image Container with Tangram for cell-type deconvolution."
    }
   },
```

**Comment by SabrinaRichter** [comment]
changed capitalization and added Cellpose to wordlist, thumbnail is already done and also appears in my html version :-)

**Comment by giovp** [comment]
looks great! thanks again will merge!

**Pull request merged**